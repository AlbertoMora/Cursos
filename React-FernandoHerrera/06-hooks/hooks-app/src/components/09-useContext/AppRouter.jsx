import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import LoginScreen from "./LoginScreen";
import NavBar from "./NavBar";

const AppRouter = () => {
    return (
        <Router>
            <>
                <NavBar />
                <Switch>
                    <Route path="/" exact component={HomeScreen} />
                    <Route path="/about" component={AboutScreen} />
                    <Route path="/login" component={LoginScreen} />
                    <Route component={HomeScreen} />
                </Switch>
            </>
        </Router>
    );
};

export default AppRouter;
