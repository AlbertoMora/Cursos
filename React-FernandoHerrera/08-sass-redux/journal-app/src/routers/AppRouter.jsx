import { firebase } from "../firebase/firebaseConfig";
import React, { useEffect, useState } from "react";
import { BrowserRouter as Router, Redirect, Switch } from "react-router-dom";
import { useDispatch } from "react-redux";

import JournalScreen from "../components/journal/JournalScreen";
import AuthRouter from "./AuthRouter";
import { login } from "../actions/auth";
import PublicRoute from "./PublicRoute";
import PrivateRoute from "./PrivateRoute";
import { startLoadingNotes } from "../actions/notes";

const AppRouter = () => {
    const dispatch = useDispatch();
    const [checking, setChecking] = useState(true);
    const [isAuth, setIsAuth] = useState(false);

    useEffect(() => {
        firebase.auth().onAuthStateChanged((user) => {
            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsAuth(true);
                dispatch(startLoadingNotes(user.uid));
            } else {
                setIsAuth(false);
            }
            setChecking(false);
        });
    }, [dispatch, setChecking]);

    if (checking) {
        return <h1>Please, wait...</h1>;
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute isAuth={isAuth} path="/auth" component={AuthRouter} />
                    <PrivateRoute isAuth={isAuth} path="/" exact component={JournalScreen} />
                    <Redirect to="/auth/login" />
                </Switch>
            </div>
        </Router>
    );
};

export default AppRouter;
