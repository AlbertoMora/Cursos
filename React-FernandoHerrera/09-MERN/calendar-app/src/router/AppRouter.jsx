import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Switch, BrowserRouter as Router, Redirect } from 'react-router-dom';
import { startCheckingAction } from '../actions/auth';
import LoginScreen from '../componets/auth/LoginScreen';
import CalendarScreen from '../componets/calendar/CalendarScreen';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () => {
    const dispatch = useDispatch();

    const { checking, user } = useSelector((state) => state.auth);

    useEffect(() => {
        dispatch(startCheckingAction());
    }, [dispatch]);

    return checking ? (
        <h1>Please Wait...</h1>
    ) : (
        <Router>
            <div>
                <Switch>
                    <PublicRoute isAuth={!!user?.uid} path="/login" exact component={LoginScreen} />
                    <PrivateRoute isAuth={!!user?.uid} path="/" exact component={CalendarScreen} />
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    );
};

export default AppRouter;
