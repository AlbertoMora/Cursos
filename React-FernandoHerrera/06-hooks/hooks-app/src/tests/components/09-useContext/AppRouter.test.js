import React from "react";
import { mount } from "enzyme";

import AppRouter from "../../../components/09-useContext/AppRouter";
import UserContext from "../../../components/09-useContext/UserContext";

describe("Tests on <AppRouter />", () => {
    const wrapper = mount(
        <UserContext.Provider value={{}}>
            <AppRouter />
        </UserContext.Provider>
    );
    test("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
