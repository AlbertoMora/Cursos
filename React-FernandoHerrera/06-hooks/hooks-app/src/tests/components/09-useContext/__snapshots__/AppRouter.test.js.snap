// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> should render correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Navbar
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  Array [
    <h1>
      Home Screen
    </h1>,
    <hr />,
    <pre
      className="container"
    />,
  ],
]
`;
